src/DacpacTool/VersionChecker.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Threading;
+using NuGet.Common;
+using NuGet.Protocol;
+using NuGet.Protocol.Core.Types;
+
+namespace MSBuild.Sdk.SqlProj.DacpacTool
+{
+
+    public class VersionChecker
+    {
+        private readonly IConsole _console;
+        private readonly IVersionProvider _versionProvider;
+
+        public VersionChecker(IConsole console, IVersionProvider versionProvider) 
+        {
+            _versionProvider = versionProvider;
+            _console = console;
+        }
+
+        public async Task CheckForPackageUpdateAsync()
+        {
+            try
+            {
+                var logger = new NullLogger();
+
+                using var cts = new CancellationTokenSource(TimeSpan.FromSeconds(5));
+
+                using var cache = new SourceCacheContext();
```

Comment:
Hmm, how does this cache work? I'm a bit worried that this might end-up going to NuGet.org every time we run the `DacpacTool`, which is essentially on every build. That has the potential of significantly slowing down build times which is in the developers inner loop. Could we perhaps cache something ourselves or is there some prior art that we can look at?

Reply:
I was thinking using the latest release info from GitHub might be more lightweight, maybe combined with some caching?

Reply:
https://api.github.com/repos/rr-wfm/MSBuild.Sdk.SqlProj/releases/latest

Reply:
No very well documented, but as I read it the cache has a 30 minute life time

Reply:
I also set a 5 second timeout (could maybe be made shorter) for the version check.

Reply:
@jmezach I have refactored the check to use the GitHub API and only call it once per day